### PR TITLE
Few improvements about localization

### PR DESCRIPTION
**Pull request by framawiki**

### CODE REVIEW

File: huggle/loadingform.cpp [hunk 1]
```diff
@@ -43,6 +43,11 @@ void LoadingForm::Info(QString text)
     this->ui->label_2->setText(text);
 }
 
+void LoadingForm::AbortLabel(QString text)
```
Comment (benapetr) [inline huggle/loadingform.cpp]: What is this function for, it's not called anywhere? If you want to translate the button just put the code to constructor of class

Comment (framawiki) [inline huggle/loadingform.cpp]: I just recopied the `Info` above.
How can we change the label of an object from an other file/class without using an intermediary function ?

File: huggle/Localization/en.xml [hunk 2]
```diff
@@ -258,7 +260,9 @@
   <string name="login-progress-user">Checking user configuration for $1</string>
   <string name="login-progress-user-info">Checking user info for $1</string>
   <string name="login-progress-whitelist">Retrieving user whitelist...</string>
+  <string name="login-projets">Projects</string>
```
Comment (benapetr) [inline huggle/Localization/en.xml]: typo? "projets"

File: huggle/loadingform.cpp (hunk 1 above)
Comment (benapetr) [inline huggle/loadingform.cpp]: The info function is used to distribute the info to both loading form and login, not to localize any stuff.

I guess that what you want is to call

```
this->ui->pushButton->setText(_l("login-abort"));
```

just insert that code to the constructor (on bottom of `LoadingForm::LoadingForm(QWidget *parent) : QDialog(parent), ui(new Ui::LoadingForm)
`)

Comment (framawiki) [inline huggle/loadingform.cpp]: that's what I wanted, thanks

File: huggle/Localization/en.xml [hunk 3]
```diff
@@ -260,7 +260,7 @@
   <string name="login-progress-user">Checking user configuration for $1</string>
   <string name="login-progress-user-info">Checking user info for $1</string>
   <string name="login-progress-whitelist">Retrieving user whitelist...</string>
-  <string name="login-projets">Projects</string>
+  <string name="login-projets">Projets</string>
```
Comment (benapetr) [inline huggle/Localization/en.xml]: but it is PROJECTS I said you made a typo in "login-projets" and now you have the same typo in both strings

File: huggle/exceptionwindow.cpp [hunk 4]
```diff
@@ -25,10 +25,10 @@ ExceptionWindow::ExceptionWindow(Exception *e) : ui(new Ui::ExceptionWindow)
         hr = "not available (nullptr)";
     this->ui->textEdit->setText("Unfortunately Huggle has crashed. Please submit the following information "\
                           "together with details of what were you doing to https://phabricator.wikimedia.org/maniphest/task/create/?projects=Huggle"\
-                          "\n\nSystem log\n==================\n" + hr
-                          + "\n\n\n\nException details\n===========================\nError code: "
+                          "\n\n\n\nException details\n===========================\nError code: "
                           + QString::number(e->ErrorCode) + "\nReason: "
-                          + e->Message + "\nSource: " + e->Source + "\nStack trace:\n" + e->StackTrace);
+                          + e->Message + "\n\nSource: " + e->Source + "\nStack trace:\n" + e->StackTrace
+                          + "\n\n\n\n===========================\nSystem log\n=======\n" + hr);
```
Comment (framawiki) [inline huggle/exceptionwindow.cpp]: Just changes the order of the lines, so Reason is at the top.

**Comment by benapetr** [comment]
OK, I don't really have a problem with changes to exception window but I would prefer if we had like separate pull requests for unrelated changes.

Anyway there are still some things that need to be sorted out before I can merge this, the "Projects" button is now completely broken, not only it says "Projets" which isn't OK, it also doesn't really work with the arrows at all.

BTW from quick observation, it seems already translated in master, so what's the point in translating it again here.

**Comment by benapetr** [comment]
I mean the arrows are working fine, but this code:

```
void Huggle::Login::on_pushButton_2_clicked()
{
    if (this->ui->tableWidget->isVisible())
    {
        this->ui->pushButton_2->setText(_l("projects") + " >>");
        this->ui->tableWidget->setVisible(false);
        hcfg->SystemConfig_Multiple = false;
    } else
    {
        hcfg->SystemConfig_Multiple = true;
        this->ui->pushButton_2->setText(_l("projects") + " <<");
        this->ui->tableWidget->setVisible(true);
        if (this->height() < 460)
            this->resize(this->width(), 480);
    }
}
```

Changes it from Projets back to Projects with arrow. So if it already works fine, why do you try to change it?

**Comment by benapetr** [comment]
I am also not sure if you are even using all these new strings, just a hint, there is a tool just run:
```
./build/localization.py
```

it will give you a list of all localization keys that are not being used

**Comment by benapetr** [comment]
Just please don't remove unused translation we use them for 2 purposes:

* To track what is still missing in source code
* Reuse the existing strings that were already translated and eventually use them somewhere else

These are already translated strings, so pls don't remove them

**Comment by framawiki** [comment]
- The button works now, label is fixed for me. Can you verify ?
- It is noted for your remark about commits, I'll try the next time to separate them into different PR.
- I think that we can rename `projects` in `login-projects`, as this localization key is only used in login.cpp. Can I just replace this in all the files of the localization folder, or should I change something on translatewiki ?
- I checked with your script and did not find unused key I added.
Thanks for your comments !

File: huggle/login.cpp [hunk 5]
```diff
@@ -1356,6 +1361,16 @@ void Huggle::Login::on_labelTranslate_linkActivated(const QString &link)
     QDesktopServices::openUrl(link);
 }
 
+void Huggle::Login::on_labelGithub_linkActivated(const QString &link)
```
Comment (framawiki) [inline huggle/login.cpp]: It may be possible later to gather these functions

**Comment by benapetr** [comment]
If you rename it in en.xml then all translation will be lost for that key

**Comment by benapetr** [comment]
The problem with 1 PR with multiple unrelated changes is that it will take much more to review, and even if few individual changes are already OK I still can't merge it because some of the changes may need some more work. I will get back soon.

**Comment by benapetr** [comment]
I just talked to a random user who never saw login form before and we agreed that both Feedback and Github links should go to About form, there really is no big need to show them on login form, which should be primarily designed for login only.

At some point we might even move the translate link, but I think it's there because login form is the only one where you can actually change the language, so it's like good for people who open the combo box and see their language is missing, so it makes sense to keep it like it is.

**Comment by framawiki** [comment]
Thanks for your reviews. I'll look later.

**Pull request merged**